torch/_dynamo/eval_frame.py

```diff
@@ -37,6 +35,7 @@
 import torch.utils.checkpoint
 from torch import _guards
 from torch._subclasses import fake_tensor
+from torch.compiler import Constraint
```

Comment:
I worry a bit about circular imports but if tests are good then I'm fine

Reply:
_dynamo has its own circular dependency problem, so all _dynamo imports are lazy to avoid this problem